docs/config/index.md

```diff
@@ -101,46 +101,74 @@ Include globs for in-source test files.
 
 When defined, Vitest will run all matched files with `import.meta.vitest` inside.
 
-### deps
+### server
 
-- **Type:** `{ external?, inline?, ... }`
+- **Type:** `{ sourcemap?, transformMode?, ... }`
+- **Version:** Since Vitest 0.34.0
 
-Handling for dependencies resolution.
+Vite-Node server options.
 
-#### deps.experimentalOptimizer
+#### server.sourcemap
 
-- **Type:** `{ ssr?, web? }`
-- **Version:** Since Vitest 0.29.0
-- **See also:** [Dep Optimization Options](https://vitejs.dev/config/dep-optimization-options.html)
+- **Type:** `'inline' | boolean`
+- **Default:** `'inline'`
 
-Enable dependency optimization. If you have a lot of tests, this might improve their performance.
+Inject inline sourcemap to modules.
 
-When Vitest encounters the external library listed in `include`, it will be bundled into a single file using esbuild and imported as a whole module. This is good for several reasons:
+#### server.transformMode
 
-- Importing packages with a lot of imports is expensive. By bundling them into one file we can save a lot of time
-- Importing UI libraries is expensive because they are not meant to run inside Node.js
-- Your `alias` configuration is now respected inside bundled packages
-- Code in your tests is running closer to how it's running in the browser
+- **Type:** `{ web?, ssr? }`
 
-Be aware that only packages in `deps.experimentalOptimizer?.[mode].include` option are bundled (some plugins populate this automatically, like Svelte). You can read more about available options in [Vite](https://vitejs.dev/config/dep-optimization-options.html) docs. By default, Vitest uses `experimentalOptimizer.web` for `jsdom` and `happy-dom` environments, and `experimentalOptimizer.ssr` for `node` and `edge` environments, but it is configurable by [`transformMode`](#transformmode).
+Determine the transform method for all modules inported inside a test that matches the glob pattern. By default, relies on the environment. For example, tests with JSDOM environment will process all files with `ssr: false` flag and tests with Node environment process all modules with `ssr: true`.
```

Comment:
This is not what it does. This is description for `testTransformMode`. `server.transformMode` changes how _files_ that match the regexp will be transformed - not _test files_.

Just use the old description for this whole `server.transformMode` part